File: src/application/models/lists/Weapons.java
package application.models.lists;

import java.io.Serializable;
import java.util.List;

import application.models.Weapon;

public class Weapons extends Equipment<Weapon> implements Serializable
{

    private static int weaponListSize = 51;

    public Weapons()
    {
	super();
    }

    public Weapons(Weapons addFrom, boolean addEmptyModel)
    {
	super(addFrom);
	if (addEmptyModel)
	{
	    this.addEmptyModel(models);
	}
    }

    public Weapons(Weapons addFrom, int equipCode)
    {
	super(addFrom, equipCode);
    }

    public Weapons(Weapons addFrom)
    {
	super(addFrom);
    }

    public Weapons(List<Integer> gameIndexes)
    {
	super(gameIndexes);
    }

    public Weapons(DModelsList<Weapon> DmodelsList)
    {
	super(DmodelsList);
    }

    @Override
    public void initializeModel(List<Weapon> models)
    {
	for (int i = 0; i < weaponListSize; i++)
	{
	    Weapon w = new Weapon(i);
	    w.getValuesFromROM();
	    models.add(w);
	}
    }

    @Override
    public void saveModels()
    {
	for (Weapon w : models)
	    w.writeValuesToROM();
    }

    @Override
    public void addEmptyModel(List<Weapon> models)
    {
	Weapon emptyWeapon = new Weapon(0);
	emptyWeapon.setGameIndex(0);
	emptyWeapon.setItemCode(0);
	emptyWeapon.setName("");
	emptyWeapon.setPower(0);
	emptyWeapon.setEquipCode(255);
	models.add(0, emptyWeapon);
    }
}

File: src/application/models/lists/Weapons.kt
package application.models.lists

import application.models.Weapon
import java.io.Serializable

class Weapons : Equipment<Weapon>, Serializable
{

	constructor() : super()
	constructor(addFrom: Weapons, equipCode: Int) : super(addFrom, equipCode)
	constructor(addFrom: Weapons) : super(addFrom)
	constructor(gameIndexes: List<Int>) : super(gameIndexes)
	constructor(DmodelsList: ModelsList<Weapon>) : super(DmodelsList)

	constructor(addFrom: Weapons, addEmptyModel: Boolean) : super(addFrom)
	{
		if (addEmptyModel) this.addEmptyModel(models)
	}


	override fun initialize(models: ModelsList<Weapon>)
	{
		for (i in 0..weaponListSize - 1)
		{
			val w = Weapon(i)
			w.getValuesFromROM()
			models.add(w)
		}
	}

	override fun saveModels() = models.forEach { it.writeValuesToROM() }

	override fun addEmptyModel(models: MutableList<Weapon>)
	{
		val emptyWeapon = Weapon(0)
		emptyWeapon.gameIndex = 0
		emptyWeapon.itemCode = 0
		emptyWeapon.name = ""
		emptyWeapon.power = 0
		emptyWeapon.equipCode = 255
		models.add(0, emptyWeapon)
	}

	companion object
	{
		private val weaponListSize = 51
	}
}
